src/js/elements/mapchip/mapchip.js

```diff
@@ -0,0 +1,182 @@
+    }
+
+    get container() {
+        return this._container;
+    }
+
+    prepareDomElements() {
+        this._container = $(mapChipBlockClass)[0];
+        // this.filterArea = this.container.find(filterContentClass); // TODO should this live here?
+        this._closeButton = $(this.container).find('.filters__header_close')
+        this._descriptionArea = $('.map-options__context .map-option__context_text div');
+        this._filtersContainer = $(this.container).find(filterContentClass);
+        this._legendContainer = $(this.container).find(legendContainerClass);
+    }
+
+    prepareEvents() {
+        this._closeButton.on('click', () => this.removeMapChip());
+    }
+
+    setDescription(text) {
+        this._descriptionArea.text(text);
+    }
+
+    setTitle(indicatorTitle, selectedSubindicator) {
+
+        let label = `${indicatorTitle} (${selectedSubindicator})`;
+        if (indicatorTitle === selectedSubindicator) {
+            label = indicatorTitle;
+        }
+
+        $(this.container).find('.filters__header_name div').text(label);
+    }
+
+    hide() {
+        $(this.container).addClass('hidden');
+        // $(mapChipBlockClass).addClass('hidden');
+    }
+
+    show() {
+        $(this.container).removeClass('hidden');
+        $(this.container).show();  //webflow.js adds display:none when clicked on x
+    }
+
+    removeMapChip() {
+        this.hide();
+        this.triggerEvent('mapchip.removed', this.container);
+    }
+
+    handleChoroplethFilter(params, dataFilterModel) {
+        let defaultFilters = [];
+        // let primaryGroup = params.primaryGroup;
+        // let groups = params.groups.filter(g => g.name != primaryGroup)
+        // this.filterController.groups = groups; // TODO should pass in the dataFilterModel, should also reset
+        // this._filterController.setDataFilterModel(dataFilterModel);
+
+        // this.filterController.clearExtraFilters();
+
+        // let filtersToAdd = [];
+        // if (typeof params.filter !== 'undefined') {
+        //     filtersToAdd = filtersToAdd.concat(params.filter);
+        // } else {
+        //     let defaultFilters = this.getDefaultFilters(params.chartConfiguration);
+        //     let nonAggFilters = this.getNonAggFilters(params.groups);
+        //     filtersToAdd = defaultFilters.concat(nonAggFilters);
+        // }
+
+        // for (let i = 1; i < filtersToAdd.length; i++) {
+        //     this.filterController.addFilter(filtersToAdd[i].default);
+        // }
+
+        // let dropdowns = $(mapChipBlockClass).find(`${filterRowClass} .mapping-options__filter`);
+
+        // this.filterController.filter = new SubindicatorFilter(
+        //     this.filterArea, groups, params.indicatorTitle, this.applyFilter, dropdowns, filtersToAdd, params.childData, '.map-options__filter-row'
+        // );
+
+        // this.filterController.setAddFilterButton();
+    }
+
+    getDefaultFilters = (chartConfiguration) => {
```

Comment:
I think the filters code should live in its own model - mapchip doesn't need to know about filter logic